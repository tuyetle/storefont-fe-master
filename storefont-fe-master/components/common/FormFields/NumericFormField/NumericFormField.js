import React, { Component } from 'react';
import PropTypes from 'prop-types';
import NumberFormat from 'react-number-format';
import numeral from 'numeral';
import { isFunction } from 'lodash';
import withLabelAndMessage from '~/hocs/withLabelAndMessage/withLabelAndMessage';
import styles from './styles';

class NumericFormField extends Component {
    componentDidMount = () => {
      if (isFunction(this.props.input.onChange) && this.props.defaultValue > 0) {
        this.props.input.onChange(this.props.defaultValue);
      }
    }

    render() {
      const { error } = this.props.meta;
      const errorLabel = error ? `${styles.error}` : '';

      return (
        <NumberFormat
          id={this.props.id}
          className={`form-control ${errorLabel}`}
          thousandSeparator={numeral.localeData().delimiters.thousands}
          decimalSeparator={numeral.localeData().delimiters.decimal}
          allowNegative={this.props.allowNegative}
          placeholder={this.props.placeholder}
          {...this.props.input}
        />
      );
    }
}

NumericFormField.propTypes = {
  id: PropTypes.string.isRequired,
  input: PropTypes.object.isRequired,
  meta: PropTypes.object.isRequired,
  defaultValue: PropTypes.number,
  allowNegative: PropTypes.bool,
  placeholder: PropTypes.string,
};

NumericFormField.defaultProps = {
  allowNegative: false,
  placeholder: '',
  defaultValue: 0,
};

export default withLabelAndMessage(NumericFormField);
