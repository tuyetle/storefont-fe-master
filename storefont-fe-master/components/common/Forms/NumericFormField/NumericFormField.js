import React, { Component } from 'react';
import PropTypes from 'prop-types';
import NumberFormat from 'react-number-format';
import numeral from 'numeral';
import { isNaN, isFunction, noop } from 'lodash';
import { withField } from 'react-form';
import styles from './styles';

export class NumericFormField extends Component {
  onChange = (values) => {
    const {
      fieldApi: {
        fieldName, touched, setValue, setTouched,
      },
      onChange,
    } = this.props;
    const { floatValue } = values;
    if (!touched) setTouched(true);

    setValue(floatValue);
    if (isFunction(onChange)) {
      onChange(fieldName, floatValue);
    }
  }

  render() {
    const {
      fieldApi: { value, touched, error }, defaultValue,
      allowNegative, placeholder,
    } = this.props;
    const errorLabel = error ? `${styles.error}` : '';

    let fieldValue = 0;
    if (touched) {
      fieldValue = isNaN(value) ? '' : value;
    } else {
      fieldValue = value || defaultValue || '';
    }

    return (
      <NumberFormat
        value={fieldValue}
        onValueChange={this.onChange}
        className={`form-control ${errorLabel}`}
        thousandSeparator={numeral.localeData().delimiters.thousands}
        decimalSeparator={numeral.localeData().delimiters.decimal}
        allowNegative={allowNegative}
        placeholder={placeholder}
      />
    );
  }
}

NumericFormField.propTypes = {
  fieldApi: PropTypes.object.isRequired,
  allowNegative: PropTypes.bool,
  placeholder: PropTypes.string,
  defaultValue: PropTypes.number,
  onChange: PropTypes.func,
};

NumericFormField.defaultProps = {
  allowNegative: false,
  placeholder: '',
  defaultValue: 0,
  onChange: noop,
};

export default withField(NumericFormField);
