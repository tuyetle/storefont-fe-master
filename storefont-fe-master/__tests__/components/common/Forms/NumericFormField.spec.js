import React from 'react';
import renderer from 'react-test-renderer';
import { NumericFormField } from '~/components/common/Forms/NumericFormField/NumericFormField';

describe('(Component) NumericFormField', () => {
  let component;
  let props;
  beforeEach(() => {
    props = {
      fieldApi: {
        setValue: () => {},
        setTouched: () => {},
      },
    };

    component = <NumericFormField {...props} />;
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

