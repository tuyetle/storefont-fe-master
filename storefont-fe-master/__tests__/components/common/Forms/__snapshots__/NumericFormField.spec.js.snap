// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) NumericFormField should render conrectly 1`] = `
<input
  className="form-control "
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseUp={[Function]}
  placeholder=""
  type="text"
  value=""
/>
`;
